### PR TITLE
Delete old OBM support.

### PR DESCRIPTION
**Pull request by zenhack**
Still need to update some of the docs, but submitting this. I'm copying @naved001's checklist from #1034 here:

- [x] Remove all old code
- [x] Update the API docs
- [x] Update the config files (sample, and dev-no-hardware).

### CODE REVIEW

**Comment by zenhack** [comment]
Docs are updated. Still working out a few kinks, but this is probably ready for review.

**Comment by naved001** [comment]
I guess we should delete the deployment tests for ipmi `tests/deployment/ipmi.py`. Other than that I think you got rid of everything else. I'll do one more pass again.

File: hil/migrations/versions/02f7e9607e16_delete_legacy_obm_support.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    for driver_table in 'mock_obm', 'ipmi':
+        if engine.dialect.has_table(engine, driver_table):
+            op.drop_table(driver_table)
+
+    op.drop_constraint(u'node_obm_id_fkey', 'node', type_='foreignkey')
+    op.drop_column('node', 'obm_id')
+    op.drop_table('obm')
+
+
+def downgrade():
```
Comment (naved001) [inline hil/migrations/versions/02f7e9607e16_delete_legacy_obm_support.py]: so we won't officially support downgrading but we'll just let it be here for someone that might need it?

Comment (zenhack) [reply hil/migrations/versions/02f7e9607e16_delete_legacy_obm_support.py]: I thinks falls under the heading of #980; this is sortof status quo of writing the downgrade function but not really supporting it.

**Comment by zenhack** [comment]
@naved001, I removed the ipmi deployment tests. From the look of them they should have been breaking for a while now (we never added the extra parameters needed to node_register). Might be a good idea to just periodically run the deployment tests to catch bitrot. We normally only run them when touching obviously pertinent functionality, but probably a good idea to do a quick sanity check every now and then.

I'll try to find time to troubleshoot Travis soonish.

File: tests/integration/client_integration.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 """Integration tests for client library"""
 from hil.flaskapp import app
 from hil.client.base import ClientBase, FailedAPICallException
-from hil.errors import BadArgumentError, UnknownSubtypeError
+from hil.errors import BadArgumentError
```
Comment (naved001) [inline tests/integration/client_integration.py]: looks like we don't use UnknownSubtypeError anywhere now. Let's just delete it while we are at it.

**Comment by zenhack** [comment]
Done.

Quoting Naved Ansari (2018-09-13 09:49:21)

**Comment by zenhack** [comment]
I believe I've fixed the travis issue.

**Pull request merged**